python_modules/libraries/dagster-pyspark/dagster_pyspark/types.py

```diff
@@ -13,17 +23,476 @@
     Selector(
         {
             'csv': {
-                'path': Field(String),
-                'sep': Field(String, is_required=False),
-                'header': Field(Bool, is_required=False),
+                'path': Field(
+                    String,
+                    is_required=True,
+                    description="the path in any Hadoop supported file system.",
+                ),
+                'mode': Field(
+                    Enum(
+                        'mode',
+                        [
+                            EnumValue(
+                                'append',
+                                description="Append contents of this :class:`DataFrame` to existing data.",
+                            ),
+                            EnumValue('overwrite', description="Overwrite existing data."),
+                            EnumValue(
+                                'ignore',
+                                description="Silently ignore this operation if data already exists.",
+                            ),
+                            EnumValue(
+                                'error',
+                                description="(default case): Throw an exception if data already exists.",
+                            ),
+                            EnumValue(
+                                'errorifexists',
+                                description="(default case): Throw an exception if data already exists.",
+                            ),
+                        ],
```

Comment:
This is great!

Let's make a single instance of this mode type and reuse it.

e.g.


```
WriteMode = Enum(
                        'WriteMode',
                        [
                            EnumValue(
                                'append',
                                description="Append contents of this :class:`DataFrame` to existing data.",
                            ),
                            EnumValue('overwrite', description="Overwrite existing data."),
                            EnumValue(
                                'ignore',
                                description="Silently ignore this operation if data already exists.",
                            ),
                            EnumValue(
                                'error',
                                description="(default case): Throw an exception if data already exists.",
                            ),
                            EnumValue(
                                'errorifexists',
                                description="(default case): Throw an exception if data already exists.",
                            ),
                        ],
              )
```

Apologies for bad indent

Reply:
Done.